ompi/mpi/java/c/Makefile.am

```diff
@@ -44,7 +46,7 @@ libmpi_java_la_SOURCES = \
 	mpi_Status.c \
 	mpi_Win.c
 
-libmpi_java_la_LIBADD = $(top_builddir)/ompi/libmpi.la
+libmpi_java_la_LIBADD = -ldl $(top_builddir)/ompi/libmpi.la
```

Comment:
well, libmpi_java calls dlopen, and libmpi_java can only be a dynamic library, so I guess this part of the code is already protected somewhere in an upper level